.github/workflows/process-pipelines.yml

```diff
@@ -3,18 +3,26 @@ name: Process Pipelines
 on:
   push: {branches: main, paths: pipelines/**}
 
+env:
+  GH_TOKEN: ${{ github.token }}
+
 jobs:
   pipeprocess:
     runs-on: ubuntu-latest
-    name: Run script
     steps:
-      - uses: actions/checkout@v3
-      
-      - run: python .github/workflows/process.py
+      - name: Create new branch and run script
+        uses: actions/checkout@v3
+        with:
+          ref: main
       - run: |
+          git branch generate-file-branch-${{ github.run_id }}-${{ github.run_attempt }}
+          git push --set-upstream origin generate-file-branch-${{ github.run_id }}-${{ github.run_attempt }}
+          git checkout generate-file-branch-${{ github.run_id }}-${{ github.run_attempt }}
+          python .github/workflows/process.py
           git config --local user.email "action@github.com"
           git config --local user.name "GitHub Action"
           git status
           git add .
           git commit -m "Update pipelines_index.json"
-          git push
\ No newline at end of file
+          git push
+          gh pr create -B main -H generate-file-branch-${{ github.run_id }}-${{ github.run_attempt }} --title 'Update pipelines_index.json' --body 'Update pipelines index'
```

Comment:
nit: Looks like the `current branch` is the default value for `-H` per [the doc](https://cli.github.com/manual/gh_pr_create), so feel free to remove it.